src/hybrid_system_constructor.jl

```diff
@@ -198,6 +198,307 @@ function PSI.construct_device!(
     return
 end
 
+function PSI.construct_device!(
+    container::PSI.OptimizationContainer,
+    sys::PSY.System,
+    ::PSI.ArgumentConstructStage,
+    model::PSI.DeviceModel{T, D},
+    network_model::PSI.NetworkModel{S},
+) where {T <: PSY.HybridSystem, D <: HybridBasicDispatch, S <: PM.AbstractPowerModel}
+    devices = PSI.get_available_components(T, sys)
+    # Add Common Variables
+    PSI.add_variables!(container, PSI.ActivePowerOutVariable, devices, D())
+    PSI.add_variables!(container, PSI.ActivePowerInVariable, devices, D())
+    PSI.add_variables!(container, PSI.ReservationVariable, devices, D())
+
+    PSI.add_to_expression!(
+        container,
+        PSI.ActivePowerBalance,
+        PSI.ActivePowerInVariable,
+        devices,
+        model,
+        network_model,
+    )
+
+    PSI.add_to_expression!(
+        container,
+        PSI.ActivePowerBalance,
+        PSI.ActivePowerOutVariable,
+        devices,
+        model,
+        network_model,
+    )
+
+    PSI.add_feedforward_arguments!(container, model, devices)
+
+    if PSI.has_service_model(model)
```

Comment:
This model is allowed to have services right? I'm not sure how this is working with this error here and then with line 245